### PR TITLE
refine: move BIT Macro to nuttx/bits.h

### PR DESCRIPTION
**Pull request by hujun260**
## Summary
refine: move BIT Macro to nuttx/bits.h
## Impact
none
## Testing
none

### CODE REVIEW

**Review by jerpelea — APPROVED** [review]
please add commit message

File: include/nuttx/bits.h [hunk 1]
```diff
@@ -26,46 +26,47 @@
...
-#include <stdint.h>
+#include <limits.h>
 
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
 
 #ifndef BITS_PER_BYTE
-# define BITS_PER_BYTE 8
+  #define BITS_PER_BYTE CHAR_BIT
```
Comment (davids5) [inline include/nuttx/bits.h:36]:
```suggestion
#  define BITS_PER_BYTE CHAR_BIT
```

File: include/nuttx/bits.h [hunk 2]
```diff
@@ -26,46 +26,47 @@
...
+#  define BITS_PER_LONG_LONG (sizeof(unsigned long long) * BITS_PER_BYTE)
 #endif
 
-#define BIT_MASK(nr)       (UINT32_C(1) << ((nr) % BITS_PER_LONG))
+#define BIT_BYTE_MASK(nr)  (1ul << ((nr) % BITS_PER_BYTE))
+#define BIT_WORD_MASK(nr)  (1ul << ((nr) % BITS_PER_LONG))
+#define BIT_BYTE(nr)       ((nr) / BITS_PER_BYTE)
 #define BIT_WORD(nr)       ((nr) / BITS_PER_LONG)
-#define BIT_ULL_MASK(nr)   (UINT64_C(1) << ((nr) % BITS_PER_LONG_LONG))
+#define BIT_ULL_MASK(nr)   (1ull << ((nr) % BITS_PER_LONG_LONG))
```
Comment (davids5) [inline include/nuttx/bits.h:51]: Why change UINT32_C(1) and UINT64_C(1)?

**Review by davids5 — CHANGES_REQUESTED** [review]
See inline

File: include/nuttx/bits.h (hunk 2 above)
Comment (xiaoxiang781216) [reply include/nuttx/bits.h:51]: Since UL/ULL is different from UINT32_C/UINT64_C.

**Review by davids5 — APPROVED** [review]

**Pull request merged**